### DISCUSSION TITLE
[docs] useMemoizedFn cannot replace useCallback when functions called during render

### DISCUSSION THREAD

**Discussion by MeetzhDing** [alibaba/hooks#2341] [General]
In the document:  [useMemoizedFn](https://ahooks.js.org/hooks/use-memoized-fn/
)

> Hooks for persistent functions. **In theory, useMemoizedFn can be used instead of useCallback.**
>
> In some scenarios, we need to use useCallback to cache a function, but when the second parameter deps changes, the function will be regenerated, causing the function reference to change.

---
But according react rfcs [useEvent](https://github.com/reactjs/rfcs/blob/useevent/text/0000-useevent.md#functions-called-during-render-still-use-usecallback), a memoized function may case error in rendering, e.g. not rerender as expected.

I suggested to declare this situation in the document.

**Comment by liuyib** [comment]
> I suggested to declare this situation in the document.

Nice catch! I will add this situation to the docs, thx.
